routing/index_router.cpp

```diff
@@ -493,9 +493,22 @@ IRouter::ResultCode IndexRouter::CalculateSubroute(Checkpoints const & checkpoin
     return isLastSubroute ? IRouter::EndPointNotFound : IRouter::IntermediatePointNotFound;
   }
 
-  graph.SetMode(AreMwmsNear(startSegment.GetMwmId(), finishSegment.GetMwmId())
-                    ? WorldGraph::Mode::LeapsIfPossible
-                    : WorldGraph::Mode::LeapsOnly);
+  // We use leaps for cars only. Other vehicle types do not have weights in their cross-mwm sections.
+  switch (m_vehicleType)
+  {
+    case VehicleType::Pedestrian:
+    case VehicleType::Bicycle:
+      graph.SetMode(WorldGraph::Mode::NoLeaps);
+      break;
+    case VehicleType::Car:
+      graph.SetMode(AreMwmsNear(startSegment.GetMwmId(), finishSegment.GetMwmId())
+                      ? WorldGraph::Mode::LeapsIfPossible
+                      : WorldGraph::Mode::LeapsOnly);
+      break;
+    default:
```

Comment:
No need in default case, because you processed all cases. And if you'll add one more item to the VehicleType, compiler will warn you that you didn't handle it here.